crates/pop-common/src/git.rs

```diff
@@ -278,6 +285,37 @@ pub struct Release {
 	pub commit: Option<String>,
 }
 
+/// Source binary by downloading and extracting from an archive.
+///
+/// # Arguments
+/// * `url` - The url of the archive.
+/// * `contents` - The contents within the archive which are required.
+/// * `status` - Used to observe status updates.
+pub async fn from_archive(
```

Comment:
Why in the git folder and why not just move the whole sourcing module?  This code downloads a source code archive over http from GitHub, and has nothing to do with git.

We now have the same dependencies defined in both places - old location and new.

My suggestion would be to move the whole sourcing module to common as a separate PR, so that both contracts and parachains crates can then use.

Reply:
The git file has the implementation of both `Git` and `GitHub` logic, `from_archive` has not been included in the specific Git logic.
The refactor to move the whole sourcing implementation was a bit tricky due to the tight coupling of some logic with parachains. For instance, `from_github_archive`  which sources and builds the binary (https://github.com/r0gue-io/pop-cli/blob/main/crates/pop-parachains/src/up/sourcing.rs#L294). That's why, as an initial step, I decided to migrate only the necessary logic required for contracts.